File: data/code_pairs/fortran/22.f90
MODULE mo_column
  CONTAINS
    FUNCTION compute_column(nz, q, t, nproma) RESULT(r)
      INTEGER, INTENT(IN) :: nproma
      INTEGER, INTENT(IN) :: nz
      REAL, INTENT(INOUT) :: t(nz, nproma)
      REAL, INTENT(INOUT) :: q(nz, nproma)
      INTEGER :: k
      REAL :: c
      INTEGER :: r
      INTEGER :: proma

      DO proma = 1, nproma
        c = 5.345
        DO k = 2, nz
          t(k, proma) = c * k
          q(k, proma) = q(k - 1, proma) + t(k, proma) * c
        END DO
        q(nz, proma) = q(nz, proma) * c
      END DO
      r = 0  ! Assuming a meaningful return could be added here
    END FUNCTION compute_column

    SUBROUTINE compute(nz, q, nproma)
      INTEGER, INTENT(IN) :: nproma
      INTEGER, INTENT(IN) :: nz
      REAL :: t(nz, nproma)
      REAL :: q(nz, nproma)
      INTEGER :: result

      result = compute_column(nz, q, t, nproma)
      ! Further processing could be done here if needed
    END SUBROUTINE compute
END MODULE mo_column

File: data/code_pairs/cpp/22.cpp
#include <vector>

namespace mo_column {

class ColumnComputer {
public:
    static void compute_column(int nz, std::vector<std::vector<double>>& q, std::vector<std::vector<double>>& t, int nproma) {
        double c = 5.345;

        for (int proma = 0; proma < nproma; ++proma) {
            for (int k = 1; k < nz; ++k) { // Adjusted for 0-based indexing
                t[k][proma] = c * (k + 1); // Fortran k is equivalent to k + 1 in C++
                q[k][proma] = q[k - 1][proma] + t[k][proma] * c;
            }
            q[nz - 1][proma] *= c;
        }
    }

    static void compute(int nz, std::vector<std::vector<double>>& q, int nproma) {
        std::vector<std::vector<double>> t(nz, std::vector<double>(nproma, 0.0)); // Initialize t with zeros
        compute_column(nz, q, t, nproma);
    }
};

} // namespace mo_column